kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +303,47 @@ def update_dataset_from_ddf(
         metadata=metadata,
         metadata_merger=metadata_merger,
     )
+
+
+def collect_dataset_statistics(
+    store, dataset_uuid, table_name, predicates=None, frac=0.3
```

Comment:
can you add a `factory` as a kwarg, please? we started to move the interface towards `kartothek.core.factory.DatasetFactory` a while ago for reading pipelines (eliminates the need to load metadata and indices multiple times)

You cann then also use `dispatch_metapartitiosn_from_factory` together with a utils function `ensure_factory` or similar (see other read implementations

Reply:
Just added it as a kwarg for now, didn't want to remove the `store` and `dataset_uuid` parameter as default for now.